node_modules/gh-users/lib/api.js

```diff
@@ -324,6 +335,243 @@ var updateAdminStatus = module.exports.updateAdminStatus = function(ctx, userId,
+        });
+    });
+};
+
+/**
+ * Remove an uploaded user CSV file
+ *
+ * @param  {Object}         userCSV                 File object representing the uploaded CSV file as returned by express
+ * @param  {Function}       callback                Standard callback function
+ * @api private
+ */
+var _cleanUpCSVFile = function(userCSV, callback) {
+    if (userCSV && userCSV.file) {
+        fs.exists(userCSV.file, function(exists) {
+            if (exists) {
+                fs.unlink(userCSV.file, function(err) {
+                    if (err) {
+                        log().warn({'err': err, 'file': userCSV}, 'Could not remove the user import CSV file');
+                    }
+                    return callback();
+                });
+            } else {
+                return callback();
+            }
+        });
+    } else {
+        return callback();
+    }
+};
+
+/**
+ * Import a set of users
+ *
+ * @param  {Number}     app                         The id of the application to import the users into
+ * @param  {String}     authenticationStrategy      The authentication strategy for the user. One of `local` or `shibboleth`
+ * @param  {Boolean}    [forceProfileUpdate]        Whether the user information should be updated, even when other user information is already present
+ * @param  {Object[]}   users                       The users to import
+ * @param  {Function}   callback                    Standard callback function
+ * @param  {Object}     callback.err                An error object, if any
+ * @api private
+ */
+var _importUsers = function(app, authenticationStrategy, forceProfileUpdate, users, callback) {
+    if (_.isEmpty(users)) {
+        return callback();
+
+    // Add a progress log statement every 50 imported users
+    } else if (users.length % 50 === 0) {
+        log().info({
+            'authenticationStrategy': authenticationStrategy,
+            'app': app
+        }, 'Importing users from CSV. ' + users.length + ' users left to import');
+    }
+
+    // Import the next user
+    var user = users.pop();
+    _importUser(app, authenticationStrategy, forceProfileUpdate, user, function(err) {
+        // Swallow the error so that 1 bad user account doesn't prevent the others
+        // from being created. The error will have been logged already
+
+        // Recursively import the next user (if any)
+        _importUsers(app, authenticationStrategy, forceProfileUpdate, users, callback);
+    });
+};
+
+/**
+ * Import a user
+ *
+ * @param  {Number}     app                         The id of the application to import the user into
+ * @param  {String}     authenticationStrategy      The authentication strategy for the user. One of `local` or `shibboleth`
+ * @param  {Boolean}    [forceProfileUpdate]        Whether the user information should be updated, even when other user information is already present
+ * @param  {Object}     user                        The user to import
```

Comment:
It seems like this might be an array of strings, not an object? If feasible, can you enumerate what we expect at what index?